### ISSUE TITLE
Admin interface not working out-of-the-box

### ISSUE THREAD

**Issue by aracloud** [buanet/docker-iobroker#115]
Hi 

Just want to play around with your docker image. But using docker or even docker-compose I was not able to access the admin interface of iobroker after running the docker image. I just run in on my Linux box with the default env settings (no settings at all).

Is there something in missed within your README part?

**I checked the logs:**

cat iobroker.current.log 
2020-06-29 14:44:00.805  - info: host.f2bfbe1162b3 iobroker.js-controller version 3.1.4 js-controller starting
2020-06-29 14:44:00.808  - info: host.f2bfbe1162b3 Copyright (c) 2014-2020 bluefox, 2014 hobbyquaker
2020-06-29 14:44:00.808  - info: host.f2bfbe1162b3 hostname: f2bfbe1162b3, node: v10.21.0
2020-06-29 14:44:00.809  - info: host.f2bfbe1162b3 ip addresses: 172.17.0.2
2020-06-29 14:44:00.844  - info: host.f2bfbe1162b3 connected to Objects and States
2020-06-29 14:44:00.918  - info: host.f2bfbe1162b3 3 instances found
2020-06-29 14:44:00.919  - warn: host.f2bfbe1162b3 does not start any instances on this host

**Tried to access the admin interface**

$ curl -I http://localhost:8081
curl: (56) Recv failure: Connection reset by peer

**There is no web server running ??** (does the admin GUI run via web interface or is there a client software needed?)

root@iobroker:/opt/iobroker# netstat -tlpn
Aktive Internetverbindungen (Nur Server)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:9000          0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:9001          0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.11:41305        0.0.0.0:*               LISTEN      -               

Thx

**Comment by buanet** [comment]
Hi, 
please provide container log. 

Or read here: 
https://forum.iobroker.net/topic/4733/howto-anleitung-installation-iobroker-in-docker-auf-synology-diskstation/

Or here:
#100 #110 #111 

There is a known issue with hostnam change on Startup. You might test beta instead of latest...

If nothing above solves your problem, please come back to my first sentence and reopen this issue. 

Regards,
André

**Comment by aracloud** [comment]
Hi André

I tried all the combinations you mentioned above but it still does not work.

Here my docker-compose part:

```
  iobroker:
    image: buanet/iobroker:beta
    container_name: iobroker
    restart: always
    hostname: iobroker
    ports:
      - "8081:8081"
    volumes:
      - iobroker-volume:/opt/iobroker
    environment:
      - HOST_HOSTNAME=iobroker

```
und hier der container log output:

```
------------------------------------------------------------
---------------     2020-06-29 23:16:32      ---------------
------------------------------------------------------------
 
------------------------------------------------------------
-----       Welcome to your ioBroker-container!        -----
-----          Startupscript is now running.           -----
-----                Please be patient!                -----
------------------------------------------------------------
 
------------------------------------------------------------
-----              Debugging information               -----
-----                                                  -----
-----                      System                      -----
-----               arch:      x86_64                  -----
-----                                                  -----
-----                     Versions                     -----
-----               image:     v5.0.0                  -----
-----               node:      v12.18.1                -----
-----               npm:       6.14.5                  -----
-----                                                  -----
-----                       ENV                        -----
-----               SETGID:    1000                    -----
-----               SETUID:    1000                    -----
------------------------------------------------------------
 
------------------------------------------------------------
-----         Step 1 of 5: Preparing container         -----
------------------------------------------------------------
 
Nothing to do here.
 
------------------------------------------------------------
-----   Step 2 of 5: Detecting ioBroker installation   -----
------------------------------------------------------------
 
Existing installation of ioBroker detected in /opt/iobroker.
 
------------------------------------------------------------
-----   Step 3 of 5: Checking ioBroker installation    -----
------------------------------------------------------------
 
(Re)Setting folder permissions (This might take a while! Please be patient!)...
Done.
 
Fixing "sudo-bug" by replacing sudo in iobroker with gosu...
Done.
 
Looks like this is a new and empty installation of ioBroker.
Hostname needs to be updated to  iobroker...
No instances found for host "a5680705ad8b"!
Done.
 
------------------------------------------------------------
-----      Step 4 of 5: Applying special settings      -----
------------------------------------------------------------
 
Some adapters have special requirements/ settings which can be activated by the use of environment variables.
For more information take a look at readme.md on Github!
 
------------------------------------------------------------
-----          Step 5 of 5: ioBroker startup           -----
------------------------------------------------------------
 
Starting ioBroker...
 
host.iobroker check instance "system.adapter.admin.0" for host "a5680705ad8b"
host.iobroker check instance "system.adapter.discovery.0" for host "a5680705ad8b"
host.iobroker check instance "system.adapter.info.0" for host "a5680705ad8b"

```

**Comment by 0kulist** [comment]
Hi @aracloud,
I think I can help with this issue, as I struggled to find info several days ago  (meanwhile, succeeded installing domoticz with docker on my x96 Max Plus TV box).
You can look at the instances:
`iobroker list instances`
```
  system.adapter.admin.0                 : admin                 -  enabled, port: "8081", bind: 0.0.0.0, run as: admin
  system.adapter.discovery.0             : discovery             -  enabled
  system.adapter.info.0                  : info                  -  enabled

+ instance is alive
```
You'll probably see no instance alive, same as me.
To start web server (i.e. admin instance) you can try the following (as seen on some forum):
`iobroker del admin.0 && iobroker del discovery.0`
`iobroker add admin && iobroker add discovery`

After that - voila:
```
root@687187dad60a:/opt/iobroker# iobroker list instances
  system.adapter.admin.0                 : admin                 -  enabled, port: "8081", bind: 0.0.0.0, run as: admin
+ system.adapter.admin.1                 : admin                 -  enabled, port: 8081, bind: 0.0.0.0, run as: admin
  system.adapter.discovery.0             : discovery             -  enabled
+ system.adapter.discovery.1             : discovery             -  enabled
  system.adapter.info.0                  : info                  -  enabled

+ instance is alive
```

At least you can take a look.
It seems that installer can't find init system, as stated in the file /opt/iobroker/INSTALLER_INFO.txt
```

cat /opt/iobroker/INSTALLER_INFO.txt
Installer version: 2020-06-12
Installation date 2020-06-22
Platform: linux
Installed as root
init system: unknown
Autostart: false
ACL enabled: true

```
I tried with both v4.2.0 and v5.0.0 with same result, and only deleting and recreating admin instance helped.

**Comment by 0kulist** [comment]
A hint for the developer - "an unresolved known issue" should be left open, if he is willing to help the community. First place to look in case of problems. Anyway, thank you @buanet.

**Comment by buanet** [comment]
@aracloud Do you want setting up a new and empty iobroker installation? 

If yes, make sure to delete your "iobroker-volume" before starting over.
If you dont do this, there is an "old" js-controller version in it which causes an actually solved issue to reappear:

```
Looks like this is a new and empty installation of ioBroker.
Hostname needs to be updated to  iobroker...
No instances found for host "a5680705ad8b"!
Done.
```

This "Existing installation of ioBroker detected in /opt/iobroker." tells me your volume is not empty on container startup.
Please try the suggested. I`m sure it will be working!

BTW: Might be a best practice hint, try using "buanet/iobroker:v5.0.0" docker tag instead of latest or beta.

Regards, 
André

**Comment by buanet** [comment]
@0kulist ok, removing and adding adapters again would work too but this is not necessary!
I reported several ways to solve it like updating js-controller or using a fixed image version within the links.

> At least you can take a look.
> It seems that installer can't find init system, as stated in the file /opt/iobroker/INSTALLER_INFO.txt
> 
> 
> ```
> cat /opt/iobroker/INSTALLER_INFO.txt
> Installer version: 2020-06-12
> Installation date 2020-06-22
> Platform: linux
> Installed as root
> init system: unknown
> Autostart: false
> ACL enabled: true
> ```
 This is absolutely normal! A container normally does not have or need an init system because it is not a virtual machine!
In my image the startup script replaces the init system and does all the magic...

BTW: The renaming issue is solved since June 3rd with v4.2.3beta. I didn't say "an unresolved known issue" at any time. Every time I'm talking about "a known issue". So as this issue was technically solved, but still in beta, I suggested every one reporting it to test if beta version solves it (which is no problem for me and should be a obvious option for everyone when having problems). 
However, hopefully this story now ends with the yesterdays release of v5.0.0... 

Regards, 
André

**Comment by aracloud** [comment]
@buanet 

Hallo André

Now, my fresh installation works (with v5.0.0) !

**So, my docker-compose is just as simple as that:**
```
  iobroker:
    image: buanet/iobroker:v5.0.0
    container_name: iobroker
    restart: always
    hostname: iobroker
    ports:
      - "8081:8081"
    volumes:
      - iobroker-volume:/opt/iobroker
```

I'm ready now to "follow the white rabbit" and digging deeper into the "rabbit hole" of IoT for future home automation projects !

Bis bald !
ARA
